browser/brave_profile_prefs.cc

```diff
@@ -409,6 +409,8 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
 
   registry->SetDefaultPrefValue(prefs::kEnableMediaRouter, base::Value(false));
 
+  registry->RegisterBooleanPref(kEnableMediaRouterOnRestart, false);
```

Comment:
this needs to be initialized to the current value of `kEnableMediaRouter` or it will disable it on next boot